src/com/facebook/buck/jvm/java/JavacStep.java

```diff
@@ -257,9 +258,24 @@ public String getShortName() {
       Path outputDirectory,
       ExecutionContext context,
       ImmutableSortedSet<Path> buildClasspathEntries) {
-    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    final ImmutableList.Builder<String> builder = ImmutableList.builder();
 
-    javacOptions.appendOptionsToList(builder, filesystem.getAbsolutifier());
+    javacOptions.appendOptionsTo(new AbstractJavacOptions.OptionsConsumer() {
+      @Override
+      public void addOptionValue(String option, String value) {
+        builder.add("-" + option).add(value);
+      }
+
+      @Override
+      public void addFlag(String flagName) {
+        builder.add("-" + flagName);
+      }
+
+      @Override
+      public void addExtras(Collection<String> extras) {
+        builder.addAll(extras);
+      }
+    }, filesystem.getAbsolutifier());
```

Comment:
nit: the formatting here is off.  `filesystem.getAbsolutifier());` should be on a line after `},`.